sklearn/neighbors/lmnn.py

```diff
@@ -0,0 +1,1003 @@
+        If False, the identity will be used, except if ``warm_start`` is True.
+
+    n_features_out : int, optional (default=None)
+        Preferred dimensionality of the inputs after the transformation.
+        If None it is inferred from ``use_pca`` and ``L``.
+
+    n_neighbors : int, optional (default=3)
+        Number of neighbors to use as target neighbors for each sample.
+
+    algorithm : {'auto', 'ball_tree', 'kd_tree', 'brute'}, optional
+        Algorithm used to compute the nearest neighbors:
+
+        - 'ball_tree' will use :class:`BallTree`
+        - 'kd_tree' will use :class:`KDTree`
+        - 'brute' will use a brute-force search.
+        - 'auto' will attempt to decide the most appropriate algorithm
+          based on the values passed to :meth:`fit` method.
+
+        Note: fitting on sparse input will override the setting of
+        this parameter, using brute force.
+
+    max_iter : int, optional (default=200)
+        Maximum number of iterations in the optimization.
+
+    tol : float, optional (default=1e-5)
+        Convergence tolerance for the optimization.
+
+    max_constraints : int, optional (default=500000)
+        Maximum number of constraints to enforce per iteration.
+
+    use_sparse : bool, optional (default=True)
+        Whether to use a sparse matrix (default) or a dense matrix for the
+        impostor-pairs storage. Using a sparse matrix, the distance to
+        impostors is computed twice, but it is somewhat faster for larger
+        data sets than using a dense matrix. With a dense matrix, the unique
+        impostor pairs have to be identified explicitly.
+
+    max_corrections : int, optional (default=100)
+        The maximum number of variable metric corrections
+        used to define the limited memory matrix. (The limited memory BFGS
+        method does not store the full hessian but uses this many terms in an
+        approximation to it.)
+
+    verbose : int, optional (default=0)
+        If 0, no progress messages will be printed.
+        If 1, progress messages will be printed to stdout.
+        If >1, progress messages will be printed and the ``iprint``
+        parameter of :meth:`fmin_l_bfgs_b` of `scipy.optimize` will be set to
+        verbose - 2.
+
+    callback : callable, optional (default=None)
+        If not None, this function is called after every iteration of the
+        optimizer taking as arguments the current solution and the number of
+        iterations (needs `scipy` >= 0.12.0).
+
+    random_state : int or numpy.RandomState or None, optional (default=None)
+        A pseudo random number generator used to sample from the constraints.
+
+    n_jobs : int, optional (default=1)
+        The number of parallel jobs to run for neighbors search.
+        If ``-1``, then the number of jobs is set to the number of CPU cores.
+        Doesn't affect :meth:`fit` method.
+
+    Attributes
+    ----------
+    L_ : array, shape (n_features_out, n_features_in).
```

Comment:
I would push for renaming this variable (here and everywhere) to something more explicit. Maybe "metric"?